antiplag/tasks.py

```diff
@@ -1,31 +1,42 @@
 from celery import shared_task
 
-from .models import Result
+from .models import Result, Submission
 
 from nlp.text_comparison import text_comparison
 from nlp.elastic import Elastic
 
 
 THRESHOLD = 0.15
 SIMILAR_COUNT = 10
-ERROR_MESSAGE = "An error occured during similarity comparison"
 
 
-@shared_task(name='antiplag.tasks.generate_result')
-def compare_documents(documents):
+@shared_task(name="antiplag.tasks.compare_documents")
+def compare_documents(submission_id):
     """
     Compare given documents against each other and N most similar documents from elastic
     """
 
+    submission = Submission.objects.get(id=submission_id)
```

Comment:
Won't this throw `ObjectDoesNotExist`? Should it be wrapped into try block?

Reply:
Yes, you are right. It throws an error instead of returning None.